torch/utils/data/sampler.py

```diff
@@ -105,11 +105,14 @@ def num_samples(self):
     def __iter__(self):
         n = len(self.data_source)
         if self.generator is None:
-            self.generator = torch.default_generator
+            generator = torch.Generator()
+            generator.manual_seed(generator.initial_seed())
```

Comment:
...seed with a `randint`. this is fixing the same generator always.

Reply:
By "default RNG" above you mean something like ```seed = random.randint(0,2147483648)```, right?

Reply:
Would you like me to change `generator.initial_seed()` part into `random.randint(0,2147483648)`?

Reply:
into something like the following
https://github.com/pytorch/pytorch/blob/2393bab036f390c53ab8b28941830530a6702b91/torch/utils/data/dataloader.py#L341

Reply:
specifically `manual_seed(torch.empty((), dtype=torch.int64).random_().item())`